src/System.Security.Permissions/tests/SecurityElementTests.cs

```diff
@@ -0,0 +1,658 @@
+        }
+
+        [Fact]
+        public void Tag()
+        {
+            SecurityElement se = new SecurityElement("Values");
+            Assert.Equal("Values", se.Tag);
+            Assert.Equal(string.Format(CultureInfo.InvariantCulture,
+                "<Values/>{0}", Environment.NewLine),
+                se.ToString());
+            se.Tag = "abc:Name";
+            Assert.Equal("abc:Name", se.Tag);
+            Assert.Equal(string.Format(CultureInfo.InvariantCulture,
+                "<abc:Name/>{0}", Environment.NewLine),
+                se.ToString());
+            se.Tag = "Name&Address";
+            Assert.Equal("Name&Address", se.Tag);
+            Assert.Equal(string.Format(CultureInfo.InvariantCulture,
+                "<Name&Address/>{0}", Environment.NewLine),
+                se.ToString());
+            se.Tag = string.Empty;
+            Assert.Equal(string.Empty, se.Tag);
+            Assert.Equal(string.Format(CultureInfo.InvariantCulture,
+                "</>{0}", Environment.NewLine),
+                se.ToString());
+        }
+
+        [Fact]
+        public void Tag_Invalid()
+        {
+            SecurityElement se = new SecurityElement("Values");
+
+            try
+            {
+                se.Tag = "Na<me";
+                Assert.False(true);
+            }
+            catch (ArgumentException ex)
+            {
+                // Invalid element tag Nam<e
+                Assert.Equal(typeof(ArgumentException), ex.GetType());
+                Assert.Null(ex.InnerException);
+                Assert.NotNull(ex.Message);
+                Assert.True(ex.Message.IndexOf("Na<me") != -1);
+                Assert.Null(ex.ParamName);
+            }
+
+            try
+            {
+                se.Tag = "Nam>e";
+                Assert.False(true);
+            }
+            catch (ArgumentException ex)
+            {
+                // Invalid element tag Nam>e
+                Assert.Equal(typeof(ArgumentException), ex.GetType());
+                Assert.Null(ex.InnerException);
+                Assert.NotNull(ex.Message);
+                Assert.True(ex.Message.IndexOf("Nam>e") != -1);
+                Assert.Null(ex.ParamName);
+            }
+        }
+
+        [Fact]
+        public void Tag_Null()
+        {
+            SecurityElement elem = CreateElement();
+            try
+            {
+                elem.Tag = null;
+                Assert.False(true);
+            }
+            catch (ArgumentNullException ex)
+            {
+                Assert.Equal(typeof(ArgumentNullException), ex.GetType());
+                Assert.Null(ex.InnerException);
+                Assert.NotNull(ex.Message);
+                Assert.NotNull(ex.ParamName);
+                Assert.Equal("Tag", ex.ParamName);
+            }
+        }
+
+        [Fact]
+        public void Text()
+        {
+            SecurityElement elem = CreateElement();
+            elem.Text = "Miguel&S�bastien";
```

Comment:
This isn't printing well on GitHub.  Did we have a unicode file import error? (Or does github have a unicode display error?)